src/app/product-list/product/product.component.ts

```diff
@@ -0,0 +1,44 @@
+
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+import { ApidataService } from 'src/app/Services/apidata.service';
+import { CartService } from 'src/app/Services/cart.service';
+import { DataService } from 'src/app/Services/data.service';
+
+
+@Component({
+  selector: 'app-product',
+  templateUrl: './product.component.html',
+  styleUrls: ['./product.component.scss']
+})
+export class ProductComponent implements OnInit, OnChanges {
+
+  @Input() products;
```

Comment:
Please create `type` or `interface` for the product and use it here to associate applicable types with the inputs

Reply:
done